src/library_exceptions.js

```diff
@@ -364,8 +379,15 @@ var LibraryExceptions = {
 #if EXCEPTION_DEBUG
     dbg("__resumeException " + [ptrToString(ptr), exceptionLast]);
 #endif
-    if (!exceptionLast) { exceptionLast = ptr; }
-    {{{ makeThrow('ptr') }}}
+    if (!exceptionLast) { 
+      exceptionLast = 
+#if EXCEPTION_STACK_TRACES
+        new CppException(ptr);
+#else
+        ptr;
+#endif
```

Comment:
I think it would be nice to avoid repeating this.  Maybe we can find some way to encapsulate this?  Perhaps ` makeThrow` itself can become smarter to avoid needing this?

Reply:
`makeThrow` is called while both creating new exception and resuming exception with the last one. For resuming exception we will have to store the instance somewhere (`exceptionLast` in changes). I can't think of a way where `makeThrow` can allocate a new exception and also use already allocated one while resuming.

Reply:
You could introduce a separate macro for `storeException(varName)` that would at least encapsulate the

```c
      exceptionLast = 
#if EXCEPTION_STACK_TRACES
        new CppException(ptr);
#else
        ptr;
#endif
```

part.

Reply:
https://github.com/emscripten-core/emscripten/pull/19099/commits/e583f39d8bcd3e7f50ea64fcb462cb5f69878ebc